### PR TITLE
Super basic form app/'context' trail

### PR DESCRIPTION
**Pull request by luketlancaster**
- [x] Add the Clubhouse Story ID: [ch21093]

not sure about...most of this. So rip 'er up and lemme know what you think

### CODE REVIEW

File: src/js/apps/forms/form/form_app.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import { LayoutView } from 'js/views/forms/form/form_views';
+
+export default App.extend({
+  currentAppOptions() {
+    return {
+      region: this.getRegion('content'),
+    };
+  },
+  onBeforeStart() {
+    this.getRegion().startPreloader(); // ??
```
Comment (luketlancaster) [inline src/js/apps/forms/form/form_app.js]: forgotten comment. I'll remove when changes are requested

File: src/js/views/forms/form/form.scss [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+.form__context-link {
+  color: $black-60;
+  cursor: pointer;
+  margin-right: 8px;
+
+  &:hover {
+    color: color(blue);
+  }
+}
```
Comment (luketlancaster) [inline src/js/views/forms/form/form.scss]: just cripped all of this from the patient scss

File: src/js/views/forms/form/form_views.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  `,
+  triggers: {
+    'click .js-back': 'click:back',
+  },
+  onClickBack() {
+    Radio.request('history', 'go:back');
+  },
+  templateContext() {
+    return {
+      src: `${ window.location.origin }/formapp/${ this.model.id }`,
```
Comment (luketlancaster) [inline src/js/views/forms/form/form_views.js]: ? stabs in the dark here

File: test/integration/forms/form.js [hunk 4]
```diff
@@ -1,21 +1,30 @@
...
       .get('[data-nav-region]')
       .should('not.be.visible');
 
-    // TODO: replace with on-screen navigation when available
     cy
-      .get('[data-nav-region]')
-      .find('[data-patients-region] .app-nav__link')
-      .last()
-      .click({ force: true });
+      .wait(2000);
```
Comment (luketlancaster) [inline test/integration/forms/form.js]: waiting doesn't feel great, but not sure the best way to make sure the form is loaded? Should it be stubbed out with form data? Or somesuch?

File: src/js/apps/forms/form/form_app.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import App from 'js/base/app';
+import Backbone from 'backbone';
+
+import { LayoutView } from 'js/views/forms/form/form_views';
+
+export default App.extend({
+  currentAppOptions() {
```
Comment (paulfalgout) [inline src/js/apps/forms/form/form_app.js]: this is the default.  you can remove this

File: src/js/views/forms/form/form_views.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+const LayoutView = View.extend({
+  template: hbs`
+      <div class="form__context-trail">
+        <a class="js-back form__context-link">
+          {{fas "chevron-left"}}{{ @intl.forms.form.formViews.contextBackBtn }}
+        </a>
+      </div>
+      <h1>{{ name }}</h1>
+      <iframe src="{{ src }}" class="form__frame"></iframe>
```
Comment (paulfalgout) [inline src/js/views/forms/form/form_views.js]: does `src="/formapp/{{ id }}"` work?

File: test/integration/forms/form.js [hunk 7]
```diff
@@ -1,21 +1,30 @@
 context('Patient Form', function() {
   specify('routing to form', function() {
+    cy
```
Comment (paulfalgout) [inline test/integration/forms/form.js]: is this visit needed?  If so that may be a problem

File: test/integration/forms/form.js (hunk 4 above)
Comment (paulfalgout) [reply test/integration/forms/form.js]: I don't think we should test the contents of the iframe, just the src for now.  _but_ https://github.com/RoundingWell/RWell/blob/develop/cypress/support/commands.js#L42 when we do.

File: src/js/views/forms/form/form_views.js (hunk 6 above)
Comment (luketlancaster) [reply src/js/views/forms/form/form_views.js]: form loads, so yeah I think so!

File: test/integration/forms/form.js (hunk 7 above)
Comment (luketlancaster) [reply test/integration/forms/form.js]: nope, sure isn't!

File: test/integration/forms/form.js [hunk 8]
```diff
@@ -7,15 +7,21 @@ context('Patient Form', function() {
...
-      .find('[data-patients-region] .app-nav__link')
-      .last()
-      .click({ force: true });
+      .wait(2000);
 
     cy
-      .get('[data-nav-region]')
-      .should('be.visible');
+      .get('iframe')
+      .should($iframe => {
```
Comment (paulfalgout) [inline test/integration/forms/form.js]: `its('src').should('contain'` I think

File: test/integration/forms/form.js (hunk 4 above)
Comment (paulfalgout) [reply test/integration/forms/form.js]: I don't think we need the wait to check the `src`?

File: test/integration/forms/form.js (hunk 8 above)
Comment (luketlancaster) [reply test/integration/forms/form.js]: gave that a shot, but was getting a 'could not find property' of src

Comment (paulfalgout) [reply test/integration/forms/form.js]: hmm maybe its doesn't work on attributes.. how about `should('have.attr'`?

File: src/js/views/forms/form/form_views.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+const LayoutView = View.extend({
+  template: hbs`
+      <div class="form__context-trail">
+        <a class="js-back form__context-link">
+          {{fas "chevron-left"}}{{ @intl.forms.form.formViews.contextBackBtn }}
+        </a>
+      </div>
+      <h1>{{ name }}</h1>
+      <iframe height="100px" src="/formapp/{{ id }}" class="form__frame"></iframe>
```
[the remark this answers is not in the document]
Comment (luketlancaster) [reply src/js/views/forms/form/form_views.js]: ah crap. I put this in to test `its` to see if it would wok on props like this and forgot to remove

**Pull request merged**